lib/WebpackError.js

```diff
@@ -4,8 +4,81 @@
 */
 "use strict";
 
+const createHash = require("./util/createHash");
+
+const hash = Symbol();
+const meta = Symbol();
+
 module.exports = class WebpackError extends Error {
+	constructor(
+		message,
+		line = 0,
+		column = 0,
+		columnEnd = 0,
+		category = "error"
+	) {
+		// TODO: enable message assertion during construction in a commit to follow
+		// if (!message) {
+		// 	WebpackError.deprecate("message must be a non-null, non-empty value.");
+		// }
+		// TODO: enable message length limit at a commit to follow
+		// else if (message.length > 160) {
+		// 	WebpackError.deprecate("message cannot be longer than 80 characters.");
+		// }
+
+		super(message);
+
+		this[hash] = createHash("md4");
+		this[meta] = {};
+		this.category = category;
+		this.column = column;
+		this.columnEnd = columnEnd;
+		this.line = line;
+
+		// TODO: enable abstract protection at a commit to follow
+		// if (this.constructor === WebpackError) {
+		// 	const message = `A WebpackError cannot be directly constructed.`;
+		// 	WebpackError.deprecate(message);
+		// }
+	}
+
+	static deprecate(message) {
+		const err = new Error();
+		const stack = err.stack
+			.split("\n")
+			.slice(3, 4)
+			.join("\n");
+		// use process.stdout to assert the message will be displayed in
+		// environments where console has been proxied. this mimics node's
+		// util.deprecate method.
+		process.stdout.write(`DeprecationWarning: ${message}\n${stack}\n`);
+	}
+
+	get id() {
```

Comment:
Why is this needed?

Reply:
This will be important and useful for reporters in the future.